src/core/border_router/routing_manager.cpp

```diff
@@ -3665,7 +3680,12 @@ void RoutingManager::PdPrefixManager::WithdrawPrefix(void)
     mPrefix.Clear();
     mTimer.Stop();
 
-    Get<RoutingManager>().ScheduleRoutingPolicyEvaluation(kImmediately);
+    // Schedule a policy evaluation only if the Routing manager is running. We may have a case where the DHCPv6
+    // prefix is obtained and expires before the Thread interface is started.
+    if (Get<RoutingManager>().IsRunning())
+    {
+        Get<RoutingManager>().ScheduleRoutingPolicyEvaluation(kImmediately);
```

Comment:
Indeed, there is no need for this code. The OT public commit that I worked on didn't check if `mIsRunning` in `ScheduleRoutingPolicyEvaluation` and our board could hang because of that. I will remove this code, thx.